metrics/metrics.go

```diff
@@ -28,7 +28,7 @@ func (s *PullService) StartInstrumentation() {
 func (s *PullService) startHandler() {
 	http.Handle("/metrics", promhttp.Handler())
 
-	if err := http.ListenAndServe(s.pullEndpoint, nil); err != nil {
+	if err := http.ListenAndServe(s.pullEndpoint, nil); err != nil { //nolint:gosec // G114: Use of net/http serve function that has no support for setting timeouts
```

Comment:
```suggestion
        server := &http.Server{
            Addr:                      s.pullEndpoint,
            ReadTimeout:       10 * time.Second,
            WriteTimeout:       10 * time.Second,
            MaxHeaderBytes: 1 << 20,
        }
	if err := server.ListenAndServe(); err != nil {
```
something like this should fix that lint error nicely without needing to ignore it